src/Orleans.TestingHost/TestStorageProviders/FaultInjectionStorageProvider.cs

```diff
@@ -135,5 +114,30 @@ private Task InsertDelay()
             logger.Info($"ClearState for grain {grainReference} of type {grainType}");
             await realStorageProvider.ClearStateAsync(grainType, grainReference, grainState);
         }
+
+        public void Participate(ISiloLifecycle lifecycle)
+        {
+            (realStorageProvider as ILifecycleParticipant<ISiloLifecycle>).Participate(lifecycle);
```

Comment:
should not assume provider is participant.  Suggest:
(realStorageProvider as ILifecycleParticipant<ISiloLifecycle>)?.Participate(lifecycle);

Reply:
totally I thought I added that ?, but I guess I didn't